mne_features/utils.py

```diff
@@ -35,6 +35,32 @@ def triu_idx(n):
             yield pos, i, j
 
 
+@nb.jit([nb.float64[:, :, ::1](nb.float64[:, :], nb.int64, nb.int64),
+         nb.float32[:, :, ::1](nb.float32[:, :], nb.int32, nb.int32)])
+def _embed_helper(x, d, tau):
+    """ Helper function for `embed`.
+
+    Parameters
+    ----------
+    x : ndarray, shape (n_channels, n_times)
+
+    d : int
+
+    tau : int
+
+    Returns
+    -------
+    output : ndarray, shape (n_channels, n_times - (d - 1) * tau, d)
+    """
+    n_channels, n_times = x.shape
+    td_emb = np.empty((n_channels, n_times - (d - 1) * tau, d), dtype=x.dtype)
+    for j in range(n_channels):
+        for i in range(n_times - (d - 1) * tau):
+            for k in range(d):
+                td_emb[j, i, k] = x[j, i + k]
+    return td_emb
```

Comment:
see https://github.com/alphacsc/alphacsc/blob/master/alphacsc/update_d.py#L181
maybe it is faster than what you do here